lib/services/nodes-api-service.js

```diff
@@ -297,6 +282,68 @@ function nodeApiServiceFactory(
         });
     };
 
+    NodeApiService.prototype.getNodeRelations = function(id) {
+        return waterline.nodes.needByIdentifier(id)
+        .then(function (node){
+            return node.relations || [];
+        });
+    };
+
+    /**
+     * Edit the relations of a given node, delegating the updates to the given handler.
+     *
+     * @param {String} id - a node id
+     * @param {Object} body - an object with relation types as keys and arrays of target
+     *      node ids as values.
+     * @param {Function} handler - a function(node, relationType, targets) which edits the
+     *      given node's relations and updates the node
+     */
+    NodeApiService.prototype.editNodeRelations = function(id, body, handler) {
+        var self = this;
+        return waterline.nodes.needByIdentifier(id).bind({})
+        .then(function(node) {
+            this.parentNode = node;
+            return Promise.all(_.transform(body, function(result, targets, relationType) {
+                result.push(self._needTargetNodes(
+                        [{relationType: relationType, targets:targets}],
+                        relationType
+                    )
+                );
+                result.push(relationType);
+            }, [])
+            );
+        })
+        .then(function(targetRelations) {
+            var parentNode = this.parentNode;
+            handler = handler.bind(self);
+            targetRelations = _.chunk(targetRelations, 2); //divide the array into subArray chunks
+                                                           //of [[targetNodes], relationType]
+            return Promise.all(_.transform(targetRelations, function(result, relationSet) {
```

Comment:
Change `Promise.all` to `Promise.each`? As the relations is mutual and may be complex. The `Promise.all` will run all promises at same time, so there may be some conflict if two promises are handling the two nodes which have mutual-relations. But `Promise.each` will run the promise one by one, so no conflict.